skyportal/plot.py

```diff
@@ -1343,6 +1344,34 @@ def photometry_plot(obj_id, user, width=600, device="browser"):
     return bokeh_embed.json_item(tabs)
 
 
+def smoothing_function(values, window_size):
+    """
+    Smooth the input "values" using a rolling average
+    where "window_size" is the number of points to use
+    for averaging.
+    This should be the same logic as static/js/plotjs/smooth_spectra.js
+    """
+
+    if values is None or not hasattr(values, '__len__') or len(values) == 0:
+        return values
+
+    output = np.zeros(values.shape)
+    under = int((window_size + 1) // 2) - 1
+    over = int(window_size // 2)
+
+    for i, v in enumerate(values):
```

Comment:
Why not perform all this with NumPy operations?

Reply:
Just trying to make it as close to the JS function as I can.

Reply:
Looks like a call to np.convolve

Reply:
I know I can make this function more efficient, but I want to make sure the logic is the same as the smoothing on the JS side, that's why it is kind of clunky.

Reply:
Since we have no test on the JavaScript side to make sure that *it* does the right convolution, perhaps it is good to use numpy here to verify that the JS does, in fact, compute borders etc. correctly?  They should match, surely.

Reply:
I don't understand how making the python side smoothing work with a convolution going to help test the JS side. Also, the edges are not exactly what you'd get from a convolution because I use a smaller window for the edges, where in a convolution you'd likely just pad with zeros or something.

Reply:
Fine, test the outputs with fixed values then.

Reply:
Not sure what you mean...

Reply:
We have a Javascript function that is untested.  We need to test it either against known values or a known implementation of the algorithm.

Reply:
I should go and look at the code again; I thought you needed the Python implementation matching the JS one so you could compare values, but I guess I didn't understand correctly what the Python version was for.

Reply:
The reason we need two versions is that when I call the internal API for the spectrum plot it needs to already send the data for one of the copies of each spectrum as smoothed (if the smooth parameter is given in the API call). 
Once on the JS side, when a user touches the smoothing button it triggers a call to the JS smoothing function, that hopefully does the right thing. 
I couldn't find a way to trigger the JS smoothing without rendering the whole thing, at which point the data is already ingested by BokehJS.